systemFlake.nix

```
@@ -7,6 +7,7 @@
 
 , nixosConfigurations ? { }
 , sharedExtraArgs ? { }
+, defaultHostAttrs ? { }
```

I think it would be preferable if equivalence was surfaced here instead of buried within the implementation. Let's think if something along the lines is possible:

```suggestion
, defaultHostAttrs ? {
  modules = sharedModules;
  extraArgs = sharedExtraArgs;
  system = defaultSystem;
}
```